cf_xarray/units.py

```diff
@@ -4,62 +4,56 @@
 import re
 
 import pint
-from pint import (  # noqa: F401
-    DimensionalityError,
-    UndefinedUnitError,
-    UnitStrippedWarning,
-)
+from packaging.version import Version
 
 from .utils import emit_user_level_warning
 
-# from `xclim`'s unit support module with permission of the maintainers
-try:
 
-    @pint.register_unit_format("cf")
-    def short_formatter(unit, registry, **options):
-        """Return a CF-compliant unit string from a `pint` unit.
-
-        Parameters
-        ----------
-        unit : pint.UnitContainer
-            Input unit.
-        registry : pint.UnitRegistry
-            the associated registry
-        **options
-            Additional options (may be ignored)
-
-        Returns
-        -------
-        out : str
-            Units following CF-Convention, using symbols.
-        """
-        import re
-
-        # convert UnitContainer back to Unit
-        unit = registry.Unit(unit)
-        # Print units using abbreviations (millimeter -> mm)
-        s = f"{unit:~D}"
-
-        # Search and replace patterns
-        pat = r"(?P<inverse>(?:1 )?/ )?(?P<unit>\w+)(?: \*\* (?P<pow>\d))?"
-
-        def repl(m):
-            i, u, p = m.groups()
-            p = p or (1 if i else "")
-            neg = "-" if i else ""
-
-            return f"{u}{neg}{p}"
-
-        out, n = re.subn(pat, repl, s)
-
-        # Remove multiplications
-        out = out.replace(" * ", " ")
-        # Delta degrees:
-        out = out.replace("Δ°", "delta_deg")
-        return out.replace("percent", "%")
+@pint.register_unit_format("cf")
+def short_formatter(unit, registry, **options):
+    """Return a CF-compliant unit string from a `pint` unit.
+
+    Parameters
+    ----------
+    unit : pint.UnitContainer
+        Input unit.
+    registry : pint.UnitRegistry
+        The associated registry
+    **options
+        Additional options (may be ignored)
+
+    Returns
+    -------
+    out : str
+        Units following CF-Convention, using symbols.
+    """
+    # pint 0.24.1 gives this for dimensionless units
+    if unit == {"dimensionless": 1}:
+        return ""
```

Comment:
Why not `return "dimensionless"` like before? I know pint 0.24.1 is doing this now, but since this is a custom formatter, this could be used to keep it backward compatible, or?

Reply:
"dimensionless" is not recognized by udunits, so we shouldn't return it. Prior to this PR, `"{u:cf}" flagged (in the doc) as returning an invalid unit because of this (ok it may not have been so clear). This PR fixes this issue.

However, returning to the conventions, I see that the empty string is indeed not really recognized either. The canonical string for dimensionless unit should maybe be "1" instead ? With the new pint we can do that.

http://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#dimensionless-units